indico/modules/events/registration/controllers/management/description.py

```diff
@@ -0,0 +1,47 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2025 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from flask import flash
+from wtforms.fields import TextAreaField
+
+from indico.modules.events.registration.controllers.management import RHManageRegFormsBase
+from indico.modules.events.settings import EventSettingsProxy
+from indico.util.i18n import _
+from indico.web.forms.base import FormDefaults, IndicoForm
+from indico.web.forms.widgets import TinyMCEWidget
+from indico.web.util import jsonify_data, jsonify_template
+
+
+event_settings = EventSettingsProxy('registration', defaults={
+    'registration_description': ''
+})
+
+
+class RegistrationDescriptionForm(IndicoForm):
+    description = TextAreaField('Registration Description',
+                                widget=TinyMCEWidget(absolute_urls=True),
+                                render_kw={'rows': 10})
+
+
+class RHManageRegistrationDescription(RHManageRegFormsBase):
+    """Manage registration description."""
+
+    def _process(self):
+        form = RegistrationDescriptionForm(
+            obj=FormDefaults(
+                description=event_settings.get(self.event, 'registration_description')
+            )
+        )
+        if form.validate_on_submit():
+            event_settings.set(self.event, 'registration_description', form.description.data)
+            flash(_('Registration description has been saved'), 'success')
+            return jsonify_data()
+        return jsonify_template(
+            'events/registration/management/registration_description.html',
+            form=form,
+            event=self.event
+        )
```

Comment:
`return jsonify_form(form)` does everything for you w/o the need for a custom template. The only think you do not get is the info box on top, but that could go in the `description` of the field instead.

Reply:
Ok, for the `info` message I used the `message` param, but up to use if you want a different solution.